build_pros.cmake

```diff
@@ -83,13 +83,15 @@ if(MSVC)
     add_definitions(-D_CRT_SECURE_NO_WARNINGS)
 endif()
 
-if("${CMAKE_C_COMPILER_ID}" MATCHES "^(Apple)?Clang$")
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-shift-negative-value")
-endif()
-
+# Tune warnings
 if(MSVC)
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /wd4244 /wd4018 /wd4996 /wd4048 /wd4267 /wd4127")
-elseif(NOT "${CMAKE_C_COMPILER_ID}" MATCHES "^(Apple)?Clang$")
+else()
+    check_c_compiler_flag("-Wno-shift-negative-value" HAVE_W_NO_SHIFT_NEGATIVE_VALUE)
```

Comment:
Just remove this whole group of warning-disable flags:
The warnings may be useful if we actually hit them.

Reply:
Done, I left MSVC-only warning disablers as these warnings are useless and tell about integer from longer to shorter conversions